schema/validator.go

```diff
@@ -117,3 +119,43 @@ func validateManifestDescendants(r io.Reader) error {
 	}
 	return nil
 }
+
+// Autodect detects the validation type
+func Autodect(f io.ReadSeeker) (string, error) {
```

Comment:
I moved the image-tools [Autodetect function](https://github.com/opencontainers/image-tools/blame/master/image/autodetect.go#L39) here because I wanted to put the JSON validation in the iamge-spec repo.